### PR TITLE
code cleanup and lots of tests

### PR DESCRIPTION
**Pull request by sbenedicadb**

### CODE REVIEW

**Review by PravinPK — APPROVED** [review]

File: AEPMessaging/Sources/MessagingRulesEngine.swift [hunk 1]
```diff
@@ -36,19 +37,209 @@ class MessagingRulesEngine {
             return
         }
 
+        // TEMP
+        //        let rules = [testRuleString, testMultipleRuleString]
```
Comment (rymorale) [inline AEPMessaging/Sources/MessagingRulesEngine.swift:41]: nit: remove test code?

File: AEPMessaging/Sources/MessagingRulesEngine.swift [hunk 2]
```diff
@@ -36,19 +37,209 @@ class MessagingRulesEngine {
...
             // TODO: handle remote assets caching (here or in UIServices?)
-
             messagingRules.append(contentsOf: processedRule)
         }
 
         rulesEngine.replaceRules(with: messagingRules)
         Log.debug(label: MessagingConstants.LOG_TAG, "Successfully loaded \(messagingRules.count) message(s) into the rules engine.")
     }
+
+    let testRuleString = """
```
Comment (rymorale) [inline AEPMessaging/Sources/MessagingRulesEngine.swift:57]: nit: should these test json strings be removed?

**Review by rymorale — APPROVED** [review]
looks good to me. it looks like some rules test strings were left that can probably be cleaned at this point.

File: AEPMessaging/Sources/Messaging+EdgeEvents.swift [hunk 3]
```diff
@@ -213,7 +212,8 @@ extension Messaging {
     /// - Returns: a `String` containing the event datasetId for Messaging
     private func getDatasetId(forEvent event: Event? = nil) -> String? {
         guard let configuration = getSharedState(extensionName: MessagingConstants.SharedState.Configuration.NAME, event: event),
-              let datasetId = configuration.experienceEventDataset else {
+              let datasetId = configuration.experienceEventDataset
+        else {
```
[the remark this answers is not in the document]
Comment (sbenedicadb) [reply AEPMessaging/Sources/Messaging+EdgeEvents.swift:216]: i guess so - i'm pretty sure i didn't write it that way 😂

File: AEPMessaging/Sources/MessagingRulesEngine.swift (hunk 2 above)
Comment (sbenedicadb) [reply AEPMessaging/Sources/MessagingRulesEngine.swift:57]: they're in there for now, i will be using them for a test later and move them to a more appropriate location

File: AEPMessaging/Sources/MessagingRulesEngine.swift (hunk 1 above)
Comment (sbenedicadb) [reply AEPMessaging/Sources/MessagingRulesEngine.swift:41]: i will remove this eventually 👍

**Pull request merged**